### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-91

### Change description ###

Add integration tests

The update of the README will be added in a separate pull request

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/AbstractServiceBusTest.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+    private void completeReceivedMessages() {
+        for (IMessage message : messagesToComplete.values()) {
+            try {
+                serviceBusClient.completeMessage(
+                    message.getMessageId(),
+                    message.getLockToken()
+                );
+            } catch (ServiceBusException e) {
+                // do nothing - in some cases completing messages is impossible at this point,
```
Comment (lgonczar) [inline src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/AbstractServiceBusTest.java]: I'll log the exception

File: src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        );
+
+        serviceBusFeeder.sendMessage(invalidMessageContent);
+
+        messageQueueProcessor.run();
+
+        assertThat(receiveMessage()).isNull();
+
+        // wait for message lock to expire
+        Thread.sleep(testConfig.getMessageLockTimeoutInMs());
```
Comment (lgonczar) [inline src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java]: Integration tests will have their own subscription (ideally dynamically created). This subscription will have a very short message lock timeout.

File: src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    @Test
+    public void run_should_read_all_messages_from_subscription() throws Exception {
+        List<PrivateBetaRegistration> registrations = Stream
+            .generate(() -> SampleData.getSampleRegistration(testConfig.getServiceName()))
+            .limit(5)
+            .collect(Collectors.toList());
+
+        for (PrivateBetaRegistration registration : registrations) {
+            serviceBusFeeder.sendMessage(registration);
+        }
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java]: could be appended to stream I think

```
Stream
  .generate(() -> SampleData.getSampleRegistration(testConfig.getServiceName()))
  .limit(5)
  .foreach(reg -> serviceBusFeeder.sendMessage(reg));
```

File: src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+
+        messageQueueProcessor.run();
+
+        // check if there's nothing left to read from the subscription
+        assertThat(receiveMessage()).isNull();
+    }
+
+    @Test
+    public void run_should_call_email_service_for_each_valid_message() throws Exception {
+        int validMesssageCount = 3;
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java]: typo (too many 's')

Comment (rkondratowicz) [reply src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java]: also, perhaps just inline, value can't be changed without breaking the test (the get(0) .. get(2) bit)

File: src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/ServiceBusClientTest.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+        // wait for message locks to expire
+        Thread.sleep(testConfig.getMessageLockTimeoutInMs());
+
+        // verify that the only message in subscription is the not completed one
+        IMessage receivedMessage = receiveMessage();
+        assertThat(receivedMessage).isNotNull();
+        assertThat(content(receivedMessage)).isEqualTo(content(notCompleted));
+        assertThat(receivedMessage.getMessageId()).isEqualTo(notCompleted.getMessageId());
+        assertThat(receiveMessage()).isNull();
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/ServiceBusClientTest.java]: that's a lot of assertions, perhaps soft assertions would add some value? (`.as(...)` may also be useful)

Comment (lgonczar) [reply src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/ServiceBusClientTest.java]: Not here - if the first assertion breaks, we don't want other assertions to be evaluated. `.as()` is a good idea - I'll add it to appropriate places.

File: src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java (hunk 3 above)
Comment (rkondratowicz) [reply src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/MessageQueueProcessorTest.java]: :(

File: src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/ServiceBusClientTest.java (hunk 5 above)
Comment (rkondratowicz) [reply src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/ServiceBusClientTest.java]: ok

**Comment by lgonczar** [comment]
Applied the suggestions

**Pull request merged**